internal/config/config.go

```diff
@@ -62,6 +63,12 @@ type Config struct {
 	once                  sync.Once
 }
 
+// ModeAuto allows you to use an automatically generated configuration and certificate.
+const (
+	ModeAuto   = "AUTO"
```

Comment:
I think we have actually 3 options:

1. spire
2. manual register + certs
3. use  self-registration  + self-generated certs

Reply:
Renamed to `spire` and `selfsigned`

Option 1 is the default option.
Option 3 is described here: https://github.com/networkservicemesh/deployments-k8s/blob/6580b816c4e1cb2382a06a4c2f27aaa38fc04778/apps/admission-webhook-k8s/README.md. 
But  I'm not sure about option 2. I haven't tested this case and haven't provided any instructions on how this can be done.

Reply:
Option 2 is already implemented in the main branch. We need to keep the functionality work.